compiler/rustc_error_codes/src/error_codes/E0308.md

```diff
@@ -3,16 +3,19 @@ Expected type did not match the received type.
 Erroneous code example:
 
 ```compile_fail,E0308
-let x: i32 = "I am not a number!";
-//     ~~~   ~~~~~~~~~~~~~~~~~~~~
-//      |             |
-//      |    initializing expression;
-//      |    compiler infers type `&str`
-//      |
-//    type `i32` assigned to variable `x`
+fn plus_one(x: i32) -> i32 {
+    x + 1
+}
+
+plus_one("Not a number");
+//       ^^^^^^^^^^^^^^ expected `i32`, found `&str`
+
+if "Not a bool" {
+// ^^^^^^^^^^^^ expected `bool`, found `&str`
+}
 ```
 
-This error occurs when the compiler is unable to infer the concrete type of a
-variable. It can occur in several cases, the most common being a mismatch
-between two types: the type the author explicitly assigned, and the type the
-compiler inferred.
+This error occurs when an expression was used in a place where the compiler
+expected an expression of a different type. It can occur in several cases, the
+most common being when calling a function, and passing an argument which has a
+different type than the matching argument in the function declaration.
```

Comment:
Something like this maybe?

```suggestion
different type than the type declared in the function declaration.
```

Reply:
"type declared in the function declaration" sounds awkward to me

Reply:
The way it is currently sounds a little funny to me, but it's not a big deal.